src/main/scala/org/scalacheck/Gen.scala

```diff
@@ -25,6 +25,198 @@ import scala.concurrent.duration.{Duration, FiniteDuration}
+ * {{{
+ * val genSmallInt: Gen[Int] = Gen.choose(-100,100)
+ * }}}
+ *
+ * This can be used to generate different collections of zero or more small integers:
+ * {{{
+ * val genListOfInts: Gen[List[Int]] = Gen.listOf(genSmallInt)
+ *
+ * val genSeqOfInts: Gen[Seq[Int]] = Gen.someOf(-100 to 100)
+ *
+ * val genVectorOfInts: Gen[Vector[Int]] = Gen.containerOf[Vector,Int](genSmallInt)
+ *
+ * val genMap: Gen[Map[Int,Boolean]] = Gen.mapOf(Gen.zip(genSmallInt, genBool))
+ *
+ * val genOptionalInt: Gen[Option[Int]] = Gen.option(genSmallInt)
+ * }}}
+ *
+ * Or collections of one or more small integers:
+ * {{{
+ * val genListOfInts: Gen[List[Int]] = Gen.nonEmptyListOf(genSmallInt)
+ *
+ * val genSeqOfInts: Gen[Seq[Int]] = Gen.atLeastOne(-100 to 100)
+ *
+ * val genVectorOfInts: Gen[Vector[Int]] = Gen.nonEmptyContainerOf[Vector,Int](genSmallInt)
+ *
+ * val genMap: Gen[Map[Int,Boolean]] = Gen.nonEmptyMap(Gen.zip(genSmallInt, genBool))
+ *
+ * val genOptionalInt: Gen[Option[Int]] = Gen.some(genSmallInt)
+ * }}}
+ *
+ * The class methods for [[Gen]] should be familiar with those in the
+ * [[scala.collection Scala collections API]]:
+ *
+ *  - [[map]] - Apply a function to generated values
+ *  - [[flatMap]] - Apply a function that returns a generator
+ *  - [[filter]] - Use values that satisfy a predicate
+ *
+ * The [[Gen]] class also supports for-comprehensions to compose
+ * complex generators:
+ *
+ * {{{
+ * val genPerson = for {
+ *   firstName <- Gen.oneOf("Alan", "Ada", "Alonzo")
+ *   lastName <- Gen.oneOf("Lovelace", "Turing", "Church")
+ *   age <- Gen.choose(0,100) if (age >= 18)
+ * } yield Person(firstName, lastName, age)
+ * }}}
+ *
+ * Constructors and factories for generators:
+ *  - [[Gen$.const const]] - Always generates a single value
+ *  - [[Gen$.oneOf[T](t0* oneOf]] - Generate a value from a list of values
+ *  - [[Gen$.atLeastOne[T](g1* atLeastOne]] - Generate a collection with at least one value from a list
+ *  - [[Gen$.someOf[T](l* someOf]] - Generate a collection with zero or more values from a list
+ *  - [[Gen$.choose choose]] - Generate numeric values in an (inclusive) range
+ *  - [[Gen$.frequency frequency]] - Choose from multiple values with a weighted distribution
+ *
+ * Combinators of generators:
+ *  - [[Gen$.buildableOf buildableOf]] - Generates a collection with a generator
+ *  - [[Gen$.buildableOfN buildableOfN]] - Generates a collection of at most ''n'' elements
+ *  - [[Gen$.nonEmptyBuildableOf nonEmptyBuildableOf]] - Generates a non-empty collection
+ *  - [[Gen$.containerOf containerOf]] - Generates a collection with a generator
+ *  - [[Gen$.containerOfN containerOfN]] - Generates a collection of at most ''n'' elements
+ *  - [[Gen$.nonEmptyContainerOf nonEmptyContainerOf]] - Generates a non-empty collection
+ *  - [[Gen$.either either]] - Generate a disjoint union of [[scala.util.Either]]
+ *  - [[Gen$.infiniteLazyList infiniteLazyList]] - Generates an infinite lazy list
```

Comment:
I'm getting this error locally when building the docs with Scala 2.12.

    Could not find any member to link for "Gen$.infiniteLazyList".

Since we have fatal warnings enabled for 2.12, that's to be expected...  I'm just surprised it's not getting caught by the GitHub build.

Reply:
The `package` task does not run `doc`; we could add it to the build script.

Reply:
See #792

Reply:
Ok, let me rebase, then?

Reply:
Rebasing shows that the error does happen in GitHub Actions when running the `doc` task. Phew.

Reply:
I like fatal warnings, so I'll just delete this line.